e2e/backend_service_update_test.go

```diff
@@ -0,0 +1,200 @@
+	tcprID := tcproute.ID(testFlags.project, meta.GlobalKey(resourceName(name)))
+	tcprRes := tcproute.NewMutableTcpRoute(testFlags.project, tcprID.Key)
+	tcprRes.Access(func(r *networkservices.TcpRoute) {
+		r.Name = name
+		r.Rules = []*networkservices.TcpRouteRouteRule{
+			{
+				Action: &networkservices.TcpRouteRouteAction{
+					Destinations: []*networkservices.TcpRouteRouteDestination{
+						{ServiceName: serviceName},
+					},
+				},
+			},
+		}
+	})
+	tcprResource, err := tcprRes.Freeze()
+	if err != nil {
+		return nil, err
+	}
+	tcprBuilder := tcproute.NewBuilder(tcprID)
+	tcprBuilder.SetOwnership(rnode.OwnershipManaged)
+	tcprBuilder.SetState(rnode.NodeExists)
+	tcprBuilder.SetResource(tcprResource)
+	builder.Add(tcprBuilder)
+	return tcprID, nil
+}
+
+func buildBackendServiceWithLBScheme(graphBuilder *rgraph.Builder, name string, lbScheme string) (*cloud.ResourceID, error) {
+	bsID := backendservice.ID(testFlags.project, meta.GlobalKey(resourceName(name)))
+
+	bsMutResource := backendservice.NewMutableBackendService(testFlags.project, bsID.Key)
+	bsMutResource.Access(func(x *compute.BackendService) {
+		x.LoadBalancingScheme = lbScheme
+		x.Protocol = "TCP"
+		x.PortName = "http"
+		x.SessionAffinity = "NONE"
+		x.Port = 80
+		x.TimeoutSec = 30
+		x.ConnectionDraining = &compute.ConnectionDraining{}
+	})
+	bsResource, err := bsMutResource.Freeze()
+	if err != nil {
+		return nil, err
+	}
+
+	bsBuilder := backendservice.NewBuilder(bsID)
+	bsBuilder.SetOwnership(rnode.OwnershipManaged)
+	bsBuilder.SetState(rnode.NodeExists)
+	bsBuilder.SetResource(bsResource)
+
+	graphBuilder.Add(bsBuilder)
+	return bsID, nil
+}
+
+func checkTCPRoute(t *testing.T, ctx context.Context, cloud cloud.Cloud, tcprID *cloud.ResourceID) {
+	t.Helper()
+	t.Log("---- Check TCP Route ---- ")
+	_, err := cloud.TcpRoutes().Get(ctx, tcprID.Key)
+	if err != nil {
+		t.Fatalf("cloud.TCPRoutes().Get(_, %s) = %v, want nil", tcprID.Key, err)
+	}
+}
+
+func checkBackendService(t *testing.T, ctx context.Context, cloud cloud.Cloud, bsID *cloud.ResourceID, bsPort int) {
+	t.Helper()
+	t.Log("---- Check BackendService ----")
+	gotBS, err := cloud.BackendServices().Get(ctx, bsID.Key)
+	if err != nil {
+		t.Fatalf("cloud.HealthChecks().Get(_, %s) = %v, want nil", bsID.Key, err)
+	}
+
+	if gotBS.Port != int64(bsPort) {
+		t.Errorf("BackendService port mismatch, got: %d want %d", gotBS.Port, bsPort)
+	}
+}
+func TestBackendServiceUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+
+	bsName := "bs-e2e"
+	bsID, err := buildBackendServiceWithLBScheme(graphBuilder, bsName, "INTERNAL_SELF_MANAGED")
+	if err != nil {
+		t.Fatalf("buildSimpleBackendService(_, %s, _) = %v, want nil", resourceName(bsName), err)
+	}
+
+	serviceName := fmt.Sprintf("https://compute.googleapis.com/v1/projects/%s/global/backendServices/%s", testFlags.project, resourceName(bsName))
```

Comment:
You already have `bsID` here. Why do you need to create it again? Also, you could use https://pkg.go.dev/github.com/GoogleCloudPlatform/k8s-cloud-provider@v1.27.0/pkg/cloud#NewBackendServicesResourceID instead

Reply:
I need a service resource URL, it doesn't seem to be the same as bsID.